src/pages/en/reference/directives-reference.md

```diff
@@ -0,0 +1,195 @@
+>
+> e.g. `<client:only="react" />` or `<client:only="my-custom-renderer" />`
+
+## Script and Style tags
+
+These directives can be used on HTML `<script>` and `<style>` tags.
+
+### `define:vars={variables}`
+
+Pass variables into a `<script>` or `<style>` tag. Any *serializable* front matter variable is supported, including props passed to your component through `Astro.props`.
+
+>⚠️ `define:vars` cannot be used on a `<script>` tag that makes use of the `hoist` directive.
+>
+> e.g. `<script hoist type="module" define:vars={{ myVariable }}>` will not work!
+
+```astro
+---
+const foregroundColor = "rgb(221 243 228)";
+const backgroundColor = "rgb(24 121 78)";
+const message = "Astro is awsome!";
+---
+<h1>Hello</h1>
+
+<style define:vars={{ textColor: foregroundColor, backgroundColor }}>
+  h1 {
+    background-color: var(--backgroundColor);
+    color: var(--textColor);
+  }
+</style>
+
+<script define:vars={{ message }}>
+  alert(message);
+</script>
+```
+
+### `hoist`
+
+Opt-in the contents of a `<script>` tag to being bundled and processed. ESM imports work, even to npm packages.
+
+Astro detects these JavaScript client-side imports and then builds, optimizes, and adds the JavaScript to the page automatically.
+
+```astro
+// ESM import
+<script hoist type="module">
+  import './some-external-script.js';
+</script>
+```
+
+>⚠️ `hoist` Always needs to be used in conjunction with `type="module"`.
+>
+> e.g. `<script hoist type="module">`
+
+### `global`
+
+Make the contents of a `<style>` tag apply globaly on pages where the component is included by disabling Astro's CSS scoping system.
+
+This is equivalent to wrapping all of the selectors within a `<style>` tag with `:global()`.
+
+```astro
+<style global>
+  a {
+    text-decoration: none;
+  }
+  a:hover {
+    text-decoration: underline;
+  }
+</style>
+<style>
+  :global(a) {
+    text-decoration: none;
+  }
+  :global(a:hover) {
+    text-decoration: underline;
+  }
+</style>
+```
+
+## HTML elements
+
+These directives can be used on any HTML element, even the lowly `<div>`.
+
+### `set:html={html}`
+
+Inject an HTML string into an element without it being escaped.
+
+```astro
+---
+const title = "Hello <strong>World</strong>"
+---
+<h1>{title}</h1> <!-- <h1>Hello &lt;strong&gt;World&lt;/strong&gt;</h1> -->
+<h1 set:html={title} /> <!-- <h1>Hello <strong>World</strong></h1> -->
+```
+
+You can also use it on a `<Fragment>` to avoid adding a wrapper element.
+
+```astro
+---
+const cmsContent = await fetchDataFromMyCMS();
+---
+<Fragment set:html={cmsContent}>
+```
+
+### `set:text={text}`
+
+The opposite of `set:html`. `set:text` ensures that any HTML content passed to it is escaped.
+
+```astro
+---
+const potentialyDangerousContent = await fetchUserGeneratedContent();
+---
+<Fragment set:text={potentialyDangerousContent}>
+```
+
+## Everything
```

Comment:
I do love your description of "the works"! :)

But, I don't think "Everything" is a useful header. And, if I've already scanned the sidebar and found what I was looking for (script, css, framework), then I wouldn't think to keep looking and notice "Everything."

So, I'm open to suggestions for this. Two thoughts come to mind, either renaming to something like "universal" (not great, but you get what I mean) and/or moving these directives up to the TOP and starting out with, "Here's some stuff you can use on everything!" then let the page get into specific use cases.